### DISCUSSION TITLE
Providing additional description to params

### DISCUSSION THREAD

**Discussion by topherfangio** [dduugg/yard-sorbet#96] [General]
Suppose I have a method like follows:

```ruby
sig { params(value: String, algorithm: Symbol).returns(String) }
def redact(value, algorithm)
  ...
end
```

Is there a good way of providing an additional description for the params or the return type? Or do we just have to type that in the standard YARD docs?

Thanks!

**Comment by KaanOzkan** [comment]
There's no built in way of adding documentation inside a sig for Sorbet. And `yard-sorbet` focuses on making Sorbet expressions [available](https://github.com/dduugg/yard-sorbet#features) in yard documentation. In your case best path forward would be to use YARD syntax as you mentioned.

I don't really see a benefit over YARD but wanted to mention that you can also split the `sig` into multiple lines and add comments for each parameter as well:
```
sig do
  params(
    value: String, # ...
    algorithm: Symbol, # ...
  ).returns(String) # ...
end
def foo; end
```

**Reply by topherfangio** [reply]
@KaanOzkan Ok, thanks for the info!

**Comment by dduugg** [comment]
I'd be open to supporting such comments  – I don't love writing the param name three times in these cases (as a YARD annotation, sig, and method def). I would want the Shopify folks such as @KaanOzkan to be on board though.

Note that there is a fork with a [change](https://github.com/scriptdash/yard-sorbet/pull/1) that incorporates inline comments into the generated documentation, but they never attempted to upstream the change.

**Reply by KaanOzkan** [reply]
Good point @dduugg on the verbosity. I personally haven't reached for this feature yet but supporting also makes sense to me since `yard-sorbet` already exposes comments on the line before a sig as YARD documentation. 👍

**Reply by dduugg** [reply]
Ah, right, I'd forgotten about that. I'm just as happy to support having one correct way to annotate params without using verbose YARD annotations. Thanks!
